### PR TITLE
fix: core themes

### PR DESCRIPTION
**Pull request by coryrylan**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)
- [ ] If applicable, have a visual design approval

## PR Type

What kind of change does this PR introduce?

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] clarity.design website / infrastructure changes
- [ ] Other... Please describe:

## What is the new behavior?
- fixes several core theme bugs and aligns closer to the design spec
- align status colors across components and fix missing states
- refactor theme status tokens to only require a single set vs primary/secondary improving performance 
- cleanup status and alert stories to better reflect statuses
- add status demo to storybook to easily see all components using status colors

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: packages/core/src/alert/alert-actions.element.scss [hunk 1]
```diff
@@ -54,15 +54,15 @@
...
+    --border-color: var(--action-text-color);
   }
 }
 
 :host([type='default']),
 :host([type='banner']) {
   &::slotted(a) {
-    color: var(--cds-global-typography-color-100);
+    color: var(--action-text-color);
   }
```
Comment (coryrylan) [inline packages/core/src/alert/alert-actions.element.scss:66]: Pointing to the single color made theming easier to inherit

File: packages/core/src/button/base-button.element.scss [hunk 2]
```diff
@@ -82,8 +82,22 @@
...
+    filter: brightness(95%);
+    position: absolute;
+    content: '';
+    top: 0;
+    left: 0;
+    bottom: 0;
+    right: 0;
+    inset: 0; // once safari catches up positions can be deleted for inset
+    z-index: -1;
+  }
```
Comment (coryrylan) [inline packages/core/src/button/base-button.element.scss:100]: This trick I figured out lets us set a darker background color programmatically without darkening the text color preserving the high contrast. Works in all browsers.

File: packages/core/src/internal/interfaces/index.ts [hunk 3]
```diff
@@ -12,9 +12,7 @@ type ExtraTShirtSizes = 'xs' | 'xxl';
 export type TShirtSizes = BasicTShirtSizes | ExtraTShirtSizes;
 
 // STATUSES
-export type StoplightStatusTypes = 'success' | 'warning' | 'danger';
-
-export type StatusTypes = 'default' | 'info' | StoplightStatusTypes;
+export type StatusTypes = 'neutral' | 'info' | 'success' | 'warning' | 'danger';
```
Comment (coryrylan) [inline packages/core/src/internal/interfaces/index.ts:15]: Updated components to be more consistent with status types. Some had default/unknown/neutral describing similar statuses types. This doesnt change any of the default styles of the components.

File: packages/core/src/alert/alert-group.stories.ts [hunk 4]
```diff
@@ -273,6 +420,93 @@ export function bannerGroup() {
...
+              <cds-button>Link 1</cds-button>
+            </a>
+          </cds-alert-actions>
+        </cds-alert>
+      </cds-alert-group>
+    </div>
+  `;
+};
+
+export const legacyWarningAlertBanner = () => {
```
Comment (coryrylan) [inline packages/core/src/alert/alert-group.stories.ts:478]: Demo showing how to override css to make warning alert match the exact color of clr-angular

**Comment by coryrylan** [comment]
Working on fixing buttons, should be 'primary' were accidentally set to info type

**Review by mathisscott — APPROVED** [review]

**Review by hippee-lee — APPROVED** [review]

**Pull request merged**